RuddockWebsite/modules/admin/helpers.py

```diff
@@ -1,8 +1,224 @@
+        grad_year=self.grad_year,
+        uid=self.uid,
+        email=self.email,
+        membership_type=self.membership_type,
+        create_account_key=create_account_key)
+    # Email the user.
+    subject = "Welcome to the Ruddock House website!"
+    msg = email_templates.AddedToWebsiteEmail.format(self.name,
+        url_for('account.create_account',
+          create_account_key=create_account_key,
+          _external=True))
+    to = self.email
+    email_utils.sendEmail(to, msg, subject)
+    return True
+
+  def set_membership_type(self):
+    '''
+    Takes a membership description (full, social, RA, etc) and sets the
+    membership_type and membership_desc attributes. If not successful, sets
+    membership_type to None.
+    '''
+    query = text("""
+      SELECT membership_type, membership_desc_short
+      FROM membership_types
+      WHERE membership_desc = :d
+        OR membership_desc_short = :d
+      """)
+    result = g.db.execute(query, d=self.membership_desc).first()
+    if result is not None:
+      self.membership_type = result['membership_type']
+      self.membership_desc = result['membership_desc_short']
+    else:
+      self.membership_type = None
+      # Don't set self.membership_desc, so we can print what the error was later.
+    return
 
-from RuddockWebsite import common_helpers, email_utils
+class NewMemberList:
+  '''
+  Class containing all data for adding one or more new members. This class is a
+  wrapper for a list of NewMember objects and associated methods.
+  '''
+  def __init__(self, new_member_list=[]):
+    self.new_member_list = new_member_list
+
+  def __str__(self):
+    ''' Returns data formatted as a CSV string. '''
+    return '\n'.join([str(new_member) for new_member in self.new_member_list])
+
+  def validate_data(self, flash_errors=True):
+    '''
+    Returns True if all data is valid. Otherwise, flashes error message(s) if
+    requested and returns False.
+    '''
+    # Check every set of new member data, don't stop at the first error.
+    is_valid = True
+    for new_member in self.new_member_list:
+      if not new_member.validate_data(flash_errors):
+        is_valid = False
+    return is_valid
+
+  def add_members(self):
+    ''' Adds all members to the database. Assumes data to be valid. '''
+    # Keep track of which members were added and which were skipped.
+    members_added = []
+    members_skipped = []
+    for new_member in self.new_member_list:
+      if new_member.add_member():
+        members_added.append(new_member.name)
+      else:
+        members_skipped.append(new_member.name)
+    flash("{0} member(s) were successfully added and {1} member(s) were skipped.".format(len(members_added), len(members_skipped)))
+    # Email admins about added members.
+
+    to = "imss@ruddock.caltech.edu, secretary@ruddock.caltech.edu"
+    msg = email_templates.MembersAddedEmail.format(
+        '\n'.join(members_added) if len(members_added) > 0 else '(none)',
```

Comment:
Better to have it than to not, since our logging infrastructure is pretty nonexistent.